include/aws/crt/JsonObject.h

```diff
@@ -33,12 +33,12 @@ namespace Aws
             /**
              * Constructs empty JSON DOM.
              */
-            JsonObject();
+            JsonObject(Allocator *a = DefaultAllocator());
```

Comment:
Allocators are usually taken by reference, as they cannot be null. Why not do that for these objects?

Reply:
Mostly just following the existing pattern established in DefaultHostResolver, HttpConnectionManager, etc...  These are the c versions.